llvm/lib/Analysis/DXILMetadataAnalysis.cpp

```diff
@@ -42,6 +83,20 @@ void ModuleMetadataInfo::print(raw_ostream &OS) const {
   OS << "Shader Stage : " << Triple::getEnvironmentTypeName(ShaderStage)
      << "\n";
   OS << "Validator Version : " << ValidatorVersion.getAsString() << "\n";
+  for (auto MapItem : FunctionPropertyMap) {
+    MapItem.first->getReturnType()->print(OS, false, true);
+    OS << " " << MapItem.first->getName() << "(";
+    FunctionType *FT = MapItem.first->getFunctionType();
+    for (unsigned I = 0, Sz = FT->getNumParams(); I < Sz; ++I) {
+      if (I)
+        OS << ",";
+      FT->getParamType(I)->print(OS);
+    }
+    OS << ")\n";
```

Comment:
The function name is probably sufficient for debugging purposes, do we really need to go through all of this effort to print a declaration?

Reply:
> The function name is probably sufficient for debugging purposes, do we really need to go through all of this effort to print a declaration?

Changed.